### PR TITLE
fix proxy URL issue

### PR DESCRIPTION
**Pull request by rmgogogo**
fix https://github.com/kubeflow/pipelines/issues/3143

### CODE REVIEW

File: proxy/attempt-register-vm-on-proxy.sh [hunk 1]
```diff
@@ -35,19 +35,10 @@ function run-proxy-agent {
...
-  PROXY_URL=$(kubectl get configmap inverse-proxy-config -o json | jq -r ".data.ProxyUrl")
-  BACKEND_ID=$(kubectl get configmap inverse-proxy-config -o json | jq -r ".data.BackendId")
-  # If ConfigMap already exist, reuse the existing endpoint (a.k.a BACKEND_ID) and same ProxyUrl.
-  run-proxy-agent
-  exit 0
-fi
+# Don't reuse existing hostname. It means if proxy-agent got restarted,
+# it will get a new hostname.
+# https://github.com/kubeflow/pipelines/issues/3143
+# Another option is that We may try to fix it in InverseProxy server side.
```
Comment (Bobgy) [inline proxy/attempt-register-vm-on-proxy.sh:41]: I feel a little worried about the url may change very frequently.

If pod restarted, is it possible VM_ID stays the same?

Comment (rmgogogo) [reply proxy/attempt-register-vm-on-proxy.sh:41]: only if pod moved to a new VM which is not frequent. That's why we seldom hit the issue.

Comment (Bobgy) [reply proxy/attempt-register-vm-on-proxy.sh:41]: Do you mean after we remove these lines of code, url will still be the same if VM ID didn't change?
then
/lgtm

**Comment by rmgogogo** [comment]
hi Yuan, I did small change after your LGTM. will just submit as a holiday.
Feel free to leave post-submit comment.

**Pull request merged**